homeassistant/components/statistics/sensor.py

```diff
@@ -277,73 +349,91 @@ def _purge_old(self):
 
     def _next_to_purge_timestamp(self):
         """Find the timestamp when the next purge would occur."""
-        if self.ages and self._max_age:
+        if self.ages and self._samples_max_age:
             # Take the oldest entry from the ages list and add the configured max_age.
             # If executed after purging old states, the result is the next timestamp
             # in the future when the oldest state will expire.
-            return self.ages[0] + self._max_age
+            return self.ages[0] + self._samples_max_age
         return None
 
+    def _update_characteristics(self):
```

Comment:
I brought this into a helper function and tidied up the logic as was meaningful. All related testcases remain green.